src/extension.js

```diff
@@ -2,33 +2,34 @@
 //
 // SPDX-License-Identifier: GPL-2.0-or-later
 
-'use strict';
+import Gio from 'gi://Gio';
+import GObject from 'gi://GObject';
 
-const Gio = imports.gi.Gio;
-const GObject = imports.gi.GObject;
-const Gtk = imports.gi.Gtk;
-
-const Main = imports.ui.main;
-const PanelMenu = imports.ui.panelMenu;
-const PopupMenu = imports.ui.popupMenu;
-const QuickSettingsMenu = Main.panel.statusArea.quickSettings;
-const QuickSettings = imports.ui.quickSettings;
+import * as Main from 'resource:///org/gnome/shell/ui/main.js';
+import * as PopupMenu from 'resource:///org/gnome/shell/ui/popupMenu.js';
+import * as QuickSettings from 'resource:///org/gnome/shell/ui/quickSettings.js';
 
 // Bootstrap
-const Extension = imports.misc.extensionUtils.getCurrentExtension();
-const Utils = Extension.imports.shell.utils;
+import {
+    Extension,
+    gettext as _,
+    ngettext
+} from 'resource:///org/gnome/shell/extensions/extension.js';
+import * as Utils from './shell/utils.js';
 
 // eslint-disable-next-line no-redeclare
-const _ = Extension._;
-const Clipboard = Extension.imports.shell.clipboard;
-const Config = Extension.imports.config;
-const Device = Extension.imports.shell.device;
-const Keybindings = Extension.imports.shell.keybindings;
-const Notification = Extension.imports.shell.notification;
-const Input = Extension.imports.shell.input;
-const Remote = Extension.imports.utils.remote;
+import * as Clipboard from './shell/clipboard.js';
+import './config.js';
+import * as Device from './shell/device.js';
+import * as Keybindings from './shell/keybindings.js';
+import * as Notification from './shell/notification.js';
+import * as Input from './shell/input.js';
+import './utils/remote.js';
+// FIXME: As a workaround, we are using globalThis for module interop.
+const Config = globalThis.GSConnectLegacyExports.Config;
+const Remote = globalThis.GSConnectLegacyExports.Remote;
```

Comment:
This would indeed be nice to avoid, but I doubt there will be any conflict.

Reply:
There are 2 alternative options I've thought of, one would be to clone `{config,remote}.js` into `{config,remote}.mjs` (`.mjs` being the Node.js extension for ES modules), and another is to enable legacy imports in the files that need `Config`/`Remote` (each file seems to need the `searchPath` modified individually). Instead of cloning the entire files, the `.mjs` files could use legacy imports to import the legacy modules and re-export them as ESM.

I've pushed 2 commits, the 1st one enables legacy imports in the individual files and uses those for imports, the 2nd one reverts that and has `.mjs` files which re-export the contents of `config.js` and `remote.js`, based on your input I can pick either of those approaches, unless you have any other idea? I think the latter approach is probably cleanest.

Reply:
Yeah, I'd agree. It's also probably easier to remove if/when the rest of the extension is ported to ESM.